### PR TITLE
Localization for Amount, Numbers and Dates.

### LINKED ISSUE #2349: Replace the existing US-located numbers, date and amounts with locale-aware components/ methods
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
We are implementing the structure for internationalisation and localisation in Expensify.Cash.

As part of that, we need to replace the existing US-located numbers, dates, amounts and phone numbers with locale-aware components/ methods. In order to do that, you need to identify all places that display any of that information and then use the methods `numberFormat`, `timestampToRelative`, `timestampToDateTime`, `toLocalPhone`, `fromLocalPhone` of the `withLocalize` HOC we added [here](https://github.com/Expensify/Expensify.cash/pull/2208).

## Expected Result:
numbers, dates, and amounts should be locale-aware

## Actual Result:
They are US centric

## Platform:
All

**Version Number:** current
https://github.com/Expensify/Expensify/issues/152154

## Job on Upwork
view Upwork job [here](https://www.upwork.com/jobs/~01aa2b8780a38644a6)

### LINKED ISSUE #3038: LHN - Phone number accounts without name set are showing @Expensify.sms in group chats
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
@expensify.sms should not be visible in any area of the app. 

## Actual Result:
Phone number accounts without name set are showing @Expensify.sms in LHN 

## Action Performed:
1. Create and account in e.cash with a Phone number
2. Do not set a name or last name in the account. 
3. Log in as another account
4. Create a new group chat that includes the phone number account. 

## Workaround:
No need, visual issue. 

## Platform:
Where is this issue occurring?

 Web ✔️ 
 iOS
 Android
 Desktop App
 Mobile Web

**Version Number:** 1.0.50-0

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

image

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___ 
From @parasharrajat https://expensify.slack.com/archives/C01GTK53T8Q/p1621543024075700
> LHN Is showing @expensify.sms in Names.

### PR DESCRIPTION
**Pull request by parasharrajat**
Please Review @iwiznia.

### Details

- [x] https://github.com/Expensify/Expensify.cash/blob/9fbf82e8237b7d1ff416619970038cc0d5bf9252/src/pages/iou/steps/IOUAmountPage.js
- [x] https://github.com/Expensify/Expensify.cash/blob/9fbf82e8237b7d1ff416619970038cc0d5bf9252/src/pages/signin/ChangeExpensifyLoginLink.js
- [x] https://github.com/Expensify/Expensify.cash/blob/9fbf82e8237b7d1ff416619970038cc0d5bf9252/src/components/IOUBadge.js
- [x] https://github.com/Expensify/Expensify.cash/blob/a75a4077b526417c09ed07316334a707cd2feea2/src/pages/home/report/ReportActionItemDate.js
- [x] https://github.com/Expensify/Expensify.cash/blob/6fcfb3fea0a0d72ae269226e9e88402607c5fa61/src/components/ReportActionItemIOUPreview.js
- [x] When type is phone 
https://github.com/Expensify/Expensify.cash/blob/a75a4077b526417c09ed07316334a707cd2feea2/src/pages/settings/Profile/LoginField.js
- [x] https://github.com/Expensify/Expensify.cash/blob/6fcfb3fea0a0d72ae269226e9e88402607c5fa61/src/pages/DetailsPage.js
- [x] https://github.com/Expensify/Expensify.cash/blob/9f305176681385f2aa80f43de06fc6dffa7f0b83/src/pages/home/sidebar/OptionRow.js
- [x] https://github.com/Expensify/Expensify.cash/blob/9f305176681385f2aa80f43de06fc6dffa7f0b83/src/pages/home/HeaderView.js
- [x] https://github.com/Expensify/Expensify.cash/blob/9f305176681385f2aa80f43de06fc6dffa7f0b83/src/pages/home/report/ReportActionItemSingle.js

### Fixed Issues

Fixes #2349 
Fixes #3038

#### Localization #2349
1. Open E.cash
2. All the times and phone numbers should be localized.
#### Expensify SMS Domain #3038

 1. Create and account in e.cash with a Phone number
 2. Do not set a name or last name in the account.
 3. Log in as another account
 4. Create a new group chat that includes the phone number account.
 5. *@expensify.sms* should not be visible in any area of the app.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web / Desktop

https://user-images.githubusercontent.com/24370807/119722071-87054a80-be89-11eb-8a4b-16e73c7cda5b.mp4

#### Android

https://user-images.githubusercontent.com/24370807/119727713-3f35f180-be90-11eb-857a-3ff7cb9429b6.mp4

### CODE REVIEW

File: src/libs/Permissions.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ Onyx.connect({
  * @returns {Boolean}
  */
 function canUseAllBetas() {
-    return _.contains(betas, CONST.BETAS.ALL);
+    return true || _.contains(betas, CONST.BETAS.ALL);
```
Comment (parasharrajat) [inline src/libs/Permissions.js:17]: I will remove these before finalizing the PR.

**Comment by parasharrajat** [comment]
Please let me know If it's going in the right direction. I personally looked at all the pages and added changes. If I am missing some parts, please mention that. Thanks.

There are few places where we are already using localization `IOUConfirmationList`. There are few errors on the IOU details pages so I am not unable to open them. But I will check the code if changes are required there.

**Review by Jag96 — COMMENTED** [review]
This is looking good to me so far, but I'll let @iwiznia comment since he's got more context here

**Comment by iwiznia** [comment]
> There are few errors on the IOU details pages so I am not unable to open them

Oh damn, what errors? Also, make sure to regularly pull main and merge it into your branch, since it is possible some of those are already fixed.

**Comment by parasharrajat** [comment]
@iwiznia when I click Pay its was crashing the app. I have reported it and I think its being fixed. I will pull in the latest changes and check again.

File: src/pages/DetailsPage.js [hunk 2]
```diff
@@ -71,20 +74,20 @@ const DetailsPage = ({personalDetails, route, translate}) => {
                                 />
                             </View>
                             <Text style={[styles.displayName, styles.mt1, styles.mb6]} numberOfLines={1}>
-                                {details.displayName
-                                    ? details.displayName
-                                    : null}
```
Comment (iwiznia) [inline src/pages/DetailsPage.js:76]: So, before when `details.displayName` was falsy, we would render `null`, now we would render `details.displayName`, is that what we want? Are there uninteded consequences of that? Mostly, I am wondering why we had this ternary condition here if there are no other consequences.

File: src/libs/Permissions.js (hunk 1 above)
Comment (iwiznia) [reply src/libs/Permissions.js:17]: Oh, I am going to create a PR so that all betas are enabled in development, so we don't have to do this anymore

**Review by iwiznia — COMMENTED** [review]
Looking good

File: src/pages/DetailsPage.js (hunk 2 above)
Comment (parasharrajat) [reply src/pages/DetailsPage.js:76]: I need to pull latest changes and I think thus this ternary would change. Why I did this as I believe we will never has a user who don't have a handle. In other words, can you imagine chating with a user who has no name(login | phone number). So if we are ended up null then we are not doing it correctly. Anyways, this issues has been fixed recently. I will pull the changes.

Comment (parasharrajat) [reply src/pages/DetailsPage.js:76]: I have updated it so return as null when there is no displayName.

File: src/pages/DetailsPage.js [hunk 3]
```diff
@@ -76,7 +76,7 @@ const DetailsPage = ({
                             <Text style={[styles.displayName, styles.mt1, styles.mb6]} numberOfLines={1}>
                                 {details.displayName && isSMSLogin
                                     ? toLocalPhone(details.displayName)
-                                    : details.displayName}
+                                    : details.displayName || null}
```
Comment (iwiznia) [inline src/pages/DetailsPage.js:79]: This whole line is a bit crazy and precedence of operations is super hard to read... can you add some parenthesis to make this clearer please?
Like:
```suggestion
                                    : (details.displayName || null)}
```
But not sure that's what you meant.

**Comment by parasharrajat** [comment]
Ready for review. All pulled latest changes from Main.

**Comment by iwiznia** [comment]
Code looks good, can you test in all platforms and remove WIP? I will do a final review then.

**Comment by parasharrajat** [comment]
Updated. Thanks. Ready to review

**Review by iwiznia — APPROVED** [review]

**Comment by iwiznia** [comment]
@parasharrajat did you test on ios? I am not seeing the video for it.

@Jag96 do you want to do one last review?

**Comment by parasharrajat** [comment]
Yeah, I tested it, I will update the video soon.

**Comment by iwiznia** [comment]
Oh, @Jag96 is out of office, @parasharrajat if you upload the ios video test, I'll merge.

**Comment by parasharrajat** [comment]
I am having issues with my mac system but I am try to recover it and hope it will work.

**Comment by iwiznia** [comment]
Any luck with that @parasharrajat?

**Comment by parasharrajat** [comment]
@iwiznia I'll update you today.

**Review by Jag96 — COMMENTED** [review]
Looks good, just one question about the format in some places. In the Header, I see we don't show the `+` and the area code, but in other places we do (example below). Is there a reason it isn't the same in both places?

image

**Comment by iwiznia** [comment]
If I had to guess is because that's text generated in the server and not in the client. Since we don't have I18n in the server, I don't think we can fix that right now.

**Comment by parasharrajat** [comment]
Yeah. I checked the main data and see that this message is coming from backend. But I can run a search for PH number on text and replace all the instances, if you want this?

**Comment by iwiznia** [comment]
No, I don't think we want that. We will need to solve this at some point when we add internationalization to the backend.

@Jag96 any other concerns? If not, do you mind merging this?

**Review by Jag96 — APPROVED** [review]
> No, I don't think we want that. We will need to solve this at some point when we add internationalization to the backend.

👍

**Pull request merged**